gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -218,38 +219,64 @@ def run(argv=None):
       known_args.minimal_vep_alt_matching,
       counter_factory)
 
-  partitioner = variant_partition.VariantPartition()
+  partitioner = None
+  if known_args.optimize_for_large_inputs or known_args.partition_config_path:
+    partitioner = variant_partition.VariantPartition(
+        known_args.partition_config_path)
+
   beam_pipeline_options = pipeline_options.PipelineOptions(pipeline_args)
   pipeline = beam.Pipeline(options=beam_pipeline_options)
   variants = _read_variants(pipeline, known_args)
   variants |= 'FilterVariants' >> filter_variants.FilterVariants(
       reference_names=known_args.reference_names)
-  if variant_merger:
-    if known_args.optimize_for_large_inputs:
-      partitions = variants | 'PartitionVariants' >> beam.Partition(
-          partition_variants.PartitionVariants(partitioner),
-          partitioner.get_num_partitions())
-      merged = []
-      for i in xrange(partitioner.get_num_partitions()):
-        merged.append(partitions[i] | 'MergeVariants' + str(i) >>
+  if partitioner:
+    num_partitions = partitioner.get_num_partitions()
+    partitioned_variants = variants | 'PartitionVariants' >> beam.Partition(
+        partition_variants.PartitionVariants(partitioner), num_partitions)
+    variants = []
+    for i in range(num_partitions):
+      if partitioner.should_keep_partition(i):
+        variants.append(partitioned_variants[i])
+      else:
+        #  An extra dummy partition has been added for residual variants. This
+        #  extra partition will be ignored in the rest of the pipeline.
+        num_partitions -= 1
+  else:
+    # By default we don't partition data, so we have only 1 partition.
+    num_partitions = 1
+    variants = [variants]
+
+  for i in range(num_partitions):
+    suffix = ''
+    if partitioner and partitioner.get_partition_name(i):
+      suffix = '_' + partitioner.get_partition_name(i)
+    if variant_merger:
+      variants[i] |= ('MergeVariants' + suffix >>
                       merge_variants.MergeVariants(variant_merger))
-      variants = merged | 'FlattenPartitions' >> beam.Flatten()
-    else:
-      variants |= ('MergeVariants' >> merge_variants.MergeVariants(
-          variant_merger))
-  proc_variants = variants | 'ProcessVaraints' >> beam.Map(
-      processed_variant_factory.create_processed_variant).\
-    with_output_types(processed_variant.ProcessedVariant)
-  _ = (proc_variants |
-       'VariantToBigQuery' >> variant_to_bigquery.VariantToBigQuery(
-           known_args.output_table,
-           header_fields,
-           variant_merger,
-           processed_variant_factory,
-           append=known_args.append,
-           allow_incompatible_records=known_args.allow_incompatible_records,
-           omit_empty_sample_calls=known_args.omit_empty_sample_calls,
-           num_bigquery_write_shards=known_args.num_bigquery_write_shards))
+    variants[i] |= (
+        'ProcessVaraints' + suffix >>
```

Comment:
I dont think this would work for auto partitions as all partitions would have the same (empty string) suffix? In any case, I dont think having `suffix` is necessary here, so just using `i` is probably sufficient.